tests/DataValues/TimeValueCalculatorTest.php

```diff
@@ -209,4 +209,450 @@ public function testGetSecondsForPrecision( $precision, $expected ) {
+					0, 0, 0,
+					TimeValue::PRECISION_SECOND,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				-85901001029,
+				-85901001029
+			],
+			[
+				new TimeValue( // BCE calculations can differ one year from external ones
+					'-0003-01-06T10:49:31Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				-62230291200,
+				-62198755201
+			],
+			[
+				new TimeValue(
+					'+0002-07-17T02:41:22Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				-62087040000,
+				-62086953601
+			],
+			[
+				new TimeValue(
+					'+2013-03-14T12:51:02Z',
+					0, 0, 0,
+					TimeValue::PRECISION_MONTH,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				1362096000,
+				1364774399
+			],
+			[
+				new TimeValue(
+					'+7254-11-27T10:49:31Z',
+					0, 0, 0,
+					TimeValue::PRECISION_SECOND,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				166775539771,
+				166775539771
+			],
+			[
+				new TimeValue(
+					'+190134401-01-15T10:39:59Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				5999999999961600,
+				6000000000047999
+			],
+			[
+				new TimeValue(
+					'+190134401-01-15T10:39:59Z',
+					0, 0, 0,
+					TimeValue::PRECISION_SECOND,
+					TimeValue::CALENDAR_GREGORIAN
+				),
+				5999999999999999,
+				5999999999999999
+			],
+		];
+		yield $timeValuesAndLowerAndHigherTimestamps;
+	}
+
+	/**
+	 * Check that a few TimeValue values are equal to their expected getLowerTimestamp() ones.
+	 *
+	 * @dataProvider provideTimeValuesAndLowerAndHigherTimestamps
+	 */
+	public function testSpecificLowerTimestamps() {
+		$timeValuesAndLowerTimestamps = func_get_args();
+		$calculator = new TimeValueCalculator();
+		foreach ( $timeValuesAndLowerTimestamps as $timeValueAndLowerTimestamp ) {
+			$unixLowerTimestamp = $calculator->getLowerTimestamp( $timeValueAndLowerTimestamp[0] );
+			$this->assertEquals( $timeValueAndLowerTimestamp[1], $unixLowerTimestamp );
+		}
+	}
+
+	/**
+	 * Check that a few TimeValue values are equal to their expected getHigherTimestamp() ones.
+	 *
+	 * @dataProvider provideTimeValuesAndLowerAndHigherTimestamps
+	 */
+	public function testSpecificHigherTimestamps() {
+		$timeValuesAndHigherTimestamps = func_get_args();
+		$calculator = new TimeValueCalculator();
+		foreach ( $timeValuesAndHigherTimestamps as $timeValueAndHigherTimestamp ) {
+			$unixHigherTimestamp = $calculator->getHigherTimestamp( $timeValueAndHigherTimestamp[0] );
+			$this->assertEquals( $timeValueAndHigherTimestamp[2], $unixHigherTimestamp );
+		}
+	}
+
+	/**
+	 * Check that timestampWithoutSignProvider() values are ordered from earliest to latest according
+	 * to getLowerTimestamp(),
+	 *
+	 * @dataProvider provideTimestampPairs
+	 */
+	public function testOrderingLowerTimestamps() {
+		$calculator = new TimeValueCalculator();
+		$timestampPairs = func_get_args();
+		foreach ( $timestampPairs as $timestampPair ) {
```

Comment:
Similarly, `func_get_args()` and the loop should not be necessary. The generator should yield a single array of function arguments, and the test function should accept those directly. The same comment applies to the other test functions below as well.